src/spring-cloud/azext_spring_cloud/custom.py

```diff
@@ -1,1079 +1,1159 @@
-    headers = urllib3.util.make_headers(
-        basic_auth='{0}:{1}'.format(user_name, password))
-    response = http.request(
-        'GET',
-        url,
-        headers=headers,
-        preload_content=False
-    )
-    try:
-        if response.status != 200:
-            raise CLIError("Failed to connect to the server with status code '{}' and reason '{}'".format(
-                response.status, response.reason))
-        std_encoding = sys.stdout.encoding
-
-        for chunk in stream(response):
-            if chunk:
-                sys.stdout.write(chunk.decode(encoding='utf-8', errors='replace')
-                                 .encode(std_encoding, errors='replace')
-                                 .decode(std_encoding, errors='replace'))
-        response.release_conn()
-    except CLIError as e:
-        exceptions.append(e)
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+# pylint: disable=unused-argument, logging-format-interpolation, protected-access, wrong-import-order, too-many-lines
+
+import yaml   # pylint: disable=import-error
+from time import sleep
+from ._stream_utils import stream_logs
+from msrestazure.azure_exceptions import CloudError
+from msrestazure.tools import parse_resource_id
+from ._utils import _get_upload_local_file
+from knack.util import CLIError
+from .vendored_sdks.appplatform import models
+from knack.log import get_logger
+from .azure_storage_file import FileService
+from azure.cli.core.util import sdk_no_wait
+from ast import literal_eval
+from azure.cli.core.commands import cached_put
+from ._utils import _get_rg_location
+from ._utils import _get_sku_tier
+from six.moves.urllib import parse
+from threading import Thread
+from threading import Timer
+import certifi
+import urllib3
+import sys
+import urllib3.contrib.pyopenssl
+
+logger = get_logger(__name__)
+DEFAULT_DEPLOYMENT_NAME = "default"
+DEPLOYMENT_CREATE_OR_UPDATE_SLEEP_INTERVAL = 5
+APP_CREATE_OR_UPDATE_SLEEP_INTERVAL = 2
+
+# pylint: disable=line-too-long
+NO_PRODUCTION_DEPLOYMENT_ERROR = "No production deployment found, use --deployment to specify deployment or create deployment with: az spring-cloud app deployment create"
+LOG_RUNNING_PROMPT = "This command usually takes minutes to run. Add '--verbose' parameter if needed."
+
+
+def spring_cloud_create(cmd, client, resource_group, name, location=None, sku=None, no_wait=False):
+    rg_location = _get_rg_location(cmd.cli_ctx, resource_group)
+    if location is None:
+        location = rg_location
+
+    if sku is None:
+        sku = "S0"
+
+    full_sku = models.Sku(
+        name=sku,
+        tier=_get_sku_tier(sku)
+    )

```

Comment:
This model construction and the `_get_sku_tier` logic could both be moved to the `validate_sku` validator if you remove `type=str` in `_params.py`. Then here you would get the `sku` parameter directly as a `models.Sku`. It's fine, just providing another common design choice in CLI.